src/cryptography/hazmat/primitives/ciphers/aead.py

```diff
@@ -365,10 +365,11 @@ def _check_params(
         data: bytes,
         associated_data: typing.List[bytes],
     ) -> None:
-        utils._check_bytes("data", data)
+        utils._check_byteslike("data", data)
         if len(data) == 0:
             raise ValueError("data must not be zero length")
         if not isinstance(associated_data, list) or not all(
-            isinstance(x, bytes) for x in associated_data
+            isinstance(x, (bytes, bytearray, memoryview))
+            for x in associated_data
         ):
             raise TypeError("associated_data must be a list of bytes or None")
```

Comment:
the exception message is no longer accurate -- rather than doing the isinstance check in the generator expression, it'd probably be better to do a regular for-loop and then checks-byteslike on each element